### ISSUE TITLE
setting a different default branch name.

### ISSUE THREAD

**Issue by ThierryO** [ropensci/git2r#434]
Is is possible to take the default branch name for the git settings rather than hard coding it to `master`? Setting `git config --global init.defaultBranch main` has no effect when using `git2r::init()`.

https://github.com/ropensci/git2r/blob/f9425f51f9ff578442391947899d4ccf6895d91f/src/libgit2/src/repository.c#L71

**Comment by stewid** [comment]
Hi @ThierryO ,

That feature was added to libgit2 v1.1 (https://github.com/libgit2/libgit2/releases/tag/v1.1.0) so it works if you first install the latest version of libgit2 before installing git2r. However, the bundled libgit2 version is 0.28, which does not have that feature. A couple of months ago I tried to update the bundled version to 1.1, but it was something I wasn't able to fix (I don't remember exactly why), therefore I'm waiting for v1.2.
